x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/IndicesAndAliasesResolver.java

```diff
@@ -447,7 +450,8 @@ static String getPutMappingIndexOrAlias(PutMappingRequest request, Predicate<Str
                     + indexAbstraction.getType().getDisplayName()
                     + "], but a concrete index is expected"
             );
-        } else if (isAuthorized.test(concreteIndexName)) {
+            // we know this is implicit data access (as opposed to another selector) so the default selector check is correct
+        } else if (isAuthorized.test(concreteIndexName, null)) {
```

Comment:
After checking the `PutMappingRequest`, it indeed does not allow selectors (`allowSelectors=false`). Wondering if we should assert this here as a sanity check and make the `data` selector explicit here?